immutables/map.py

```diff
@@ -46,6 +46,13 @@ def map_bitindex(bitmap, bit):
 W_EMPTY, W_NEWNODE, W_NOT_FOUND = range(3)
 void = object()
 
+class _Unhashable:
+    __slots__ = ()
+    __hash__ = None
+
+_NULL = _Unhashable()
```

Comment:
I'd call it `NULL`, the module isn't imported directly anyways. Also I think you can just use `void` since this is just a marker that users never see.

Reply:
+1. Note, `void` is hashable (`object` implements `__hash__` without passing it down to subclasses), but that doesn’t matter if this is just used as a sentinel object anyway?

```
>>> foo = object()
>>> hash(foo)
9007199254740991
```

Reply:
I guess I'm being a little paranoid, but I would prefer an immediate error, which `void` wouldn't do. This is the current behavior:
```
>>> from immutables.map import Map
>>> m = Map().set(None, 1)
# do a bunch of stuff
>>> None in m
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "$prefix/lib/site-packages/immutables/map.py", line 592, in __contains__
    self.__root.find(0, map_hash(key), key)
  File "$prefix/lib/site-packages/immutables/map.py", line 157, in find
    return val_or_node.find(shift + 5, hash, key)
AttributeError: 'int' object has no attribute 'find'
```
With the patch (if somehow the user uses `_NULL` as key) it raises a `TypeError` immediately.

Whatever you decide, I can change with a button click.